core/src/main/java/jenkins/security/ExceptionTranslationFilter.java

```diff
@@ -122,9 +126,9 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 		catch (AuthenticationException | AccessDeniedException ex) {
 			handleException(request, response, chain, ex);
 		} catch (ServletException ex) {
-			if (ex.getRootCause() instanceof AuthenticationException
-					|| ex.getRootCause() instanceof AccessDeniedException) {
-				handleException(request, response, chain, (AcegiSecurityException) ex.getRootCause());
+			AcegiSecurityException securityException = containsAccessException(ex);
```

Comment:
this is a larger change.  if I handle the access denied exception but wrap it with another exception that then causes an exception the current logic is to actually show the top level exception.  wheras now it punts to the access denied exception.  whats the case for looking deeper than one level?

Reply:
The `/configureSecurity` case @Wadeck described earlier is one example, as demonstrated in this new test [here](https://github.com/jenkinsci/jenkins/blob/232edf5670ef254376444777fe016f80f8e9f9c4/test/src/test/java/jenkins/security/StackTraceSuppressionTest.java#L74). The `AccessDeniedException2` is wrapped several levels deep in `RuntimeException`. In contrast, `/manage` comes through without the deeper check as shown in this new test [here](https://github.com/jenkinsci/jenkins/blob/232edf5670ef254376444777fe016f80f8e9f9c4/test/src/test/java/jenkins/security/StackTraceSuppressionTest.java#L58). The choice here is basically to let it be caught by the unhandled exception code or the access denied exception code. It's more helpful when it's the latter.

Reply:
I imagine we can thank jelly for the wrapping there :p

Reply:
my gut tells me this is wrong, jelly just adds the servlet exception so one level which is what this code originally did.
if you are absolutely sure otherwise  then I believe the previous code loop should be removed and collapsed with this. (pretty sure both of the previous are subtypes of this exception off the top of my head)

Reply:
Caught at line 128 of ExceptionTranslationFilter when a non-administrator user goes to `/configureSecurity` by `getCause()`:
ServletException
  RuntimeException
    JellyTagException
      AccessDeniedException2

At the same point, `ex.getRootCause()` gives a RuntimeException.

Reply:
@jtnord is the answer satisfying you or do you need something else? :)

Reply:
I would argue that is a bug in the code - the jelly should be hidden behind a permission check. 
`<layout permission=Administer>...`

Reply:
looks like there is a permission check - so is this a bug against groovy views?

Reply:
I backed out the bit to check embedded exceptions.